ads/google/a4a/line-delimited-response-handler.js

```diff
@@ -22,7 +22,7 @@ import {tryParseJson} from '../../../src/json';
  * @param {!Window} win
  * @param {!Response} response
  * @param {function(string, boolean)} lineCallback
- * @private
+ * @protected
```

Comment:
I don't really understand how `@protected` applies to non-properties things. `@private` is at least explained to be available to code inside the file, but what does `@protected` do that's not just `@public`? (Obviously `@protected` properties are only available to subclasses)

Reply:
The documentation is at https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#protected. I wonder if `@protected` implicitly allows access within the same directory.

Reply:
We should figure this out.

Reply:
`@private` already does that. `@protected` might be the same as `@public` for non-class functions, in which case we should probably just remove the declaration outright.

Reply:
Closure compiler source code seems to indicate that `@public` and `@protected` are equivalent when it comes to name visibility. See lines 585 and 586 [here](https://github.com/google/closure-compiler/blob/50d41151fb0fb79cb61bf1aaecc758383f96c5f1/src/com/google/javascript/jscomp/CheckAccessControls.java#L543-L589).

Reply:
Following the rabbit hole, it seems `@package` visibility is equivalent to directory-level private. https://github.com/google/closure-compiler/blob/246ab429f96bff062941a4da8b8f2af01ef5eb0d/src/com/google/javascript/jscomp/CodingConventions.java#L368-L374

Reply:
Or the [documentation](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#package). :)

> Only code in the same directory can access names marked @package.

I originally suggested using `@package` in the Resources refactor since a lot of previously private functions suddenly became public.

Reply:
I'm happy with changing all of these `@protected` to `@private @package`.

Reply:
There are ~300 instances of `@protected` in the codebase. Would it make sense to try and change as many of them as possible to `@private @package`?

Reply:
Why would you declare `@private @package` on a property at the same time?

Reply:
Oh, `@package` implies "directory `@private`". I thought it was a modifier on top of `@private`.

Reply:
I've pushed a new commit that replaces `@protected` with `@package` across the codebase, wherever possible. PTAL.